text/0000-project-ffi-unwind.md

```diff
@@ -0,0 +1,211 @@
+  charter and instructions for how community members can monitor the group's
+  progress and/or participate.
+
+[This blog post][shepherds-3.0] explains in detail the role of the
+shepherds.
+
+## Project group roadmap and RFCs
+
+The first step of the project group is to define a **roadmap** indicating the
+planned sequence in which it will design and propose particular behaviors and
+features.  Once the project group feels it has completed work on some item in
+the roadmap, that item will be submitted to the community as an RFC or FCP.
+
+## Stabilizing unspecified "TBD" behavior
+[stabilizing-tbd]: stabilizing-unspecified-tbd-behavior
+
+We would like to be able to provide features in stable Rust without a full
+formal specification but with an informal statement of intent regarding the
+behavior. These features would be considered "unspecified behavior" (rather
+than "undefined behavior"), and their behavior would therefore be subject to
+some change, but such a change would be guaranteed not to violate the project
+group's stated intent for the feature.
+
+Internally, a project group should refer to such unspecified behavior as "TBD",
+to indicate that refining the specification of the behavior is within the scope
+of the project. Outside of the context of the project group, such as in general
+reference material for the language, the behavioral intent as determined by the
+project group should be described, but will not imply any special status
+compared to other instances of unspecified behavior in the language.
+
+## Details of the FFI-unwind project group
+
+[Repository][ffi-unwind project]
+
+Initial shepherds:
+
+* [acfoltzer (Adam)](https://github.com/acfoltzer)
+* [batmanaod (Kyle)](https://github.com/batmanaod)
+
+Lang team liaisons:
+
+* [nikmoatsakis (Niko)](https://github.com/nikmoatsakis)
+* [joshtriplett (Josh)](https://github.com/joshtriplett)
+
+### Charter
+[charter]: #charter
+
+The FFI-unwind project group has the following initial scope:
+
+* to define the details of the "C unwind" ABI on major Tier 1 platforms
+* in particular, to define with sufficient detail to enable the use cases
+  described in the Motivation section of this RFC
+  
+Certain elements are definitively out of scope:
+
+* The group does not intend to consider mechanisms to enable "interop"
+  between Rust panics and exceptions from other languges. For example,
+  we do not intend to permit Rust code to catch C++ exceptions, though
+  we will have to consider what happens when a C++ exception unwinds
+  past a `catch_unwind` boundary.
+
+### Constraints and considerations
+
+In its work, the project-group should consider various constraints and
+considerations:
+
+* The possibility that C++ may adopt new unwinding mechanisms in the future.
+* The possibility that Rust may alter its unwinding mechanism in the future --
+  in particular, the project group must not propose a design that would
+  constrain Rust's unwinding implementation.
```

Comment:
```suggestion
  constrain Rust's unwinding implementation, including on major Tier 1 platforms.
```

Reply:
How about "...implementation on current or future platforms"? Calling out Tier 1 here seems to imply that perhaps they _should_ be treated differently in some way, which isn't the intent of this section.

Reply:
I used "Tier 1" here because it is specifically called out in the Charter. I'd be fine with "on current or future platforms" but I've clarified previously how I feel about Tier 1 platforms as well on Zulip so y'all can take that into account in the PG.

Reply:
Sorry, I don't remember comments from you about Tier 1 platforms, and I can't find them with the search function! Care to reiterate?

Reply:
I think we should be free to change Rut's unwinding implementations on Tier 1 platforms as well. Otherwise the freedom suggested here wouldn't mean much in practice.

Reply:
>  constrain Rust's unwinding implementation, including on major Tier 1 platforms.

@Centril I think the "including on major Tier 1 platforms" is unnecessary, the intent is to avoid constraining the Rust unwind implementation on any platform, not only Tier 1, and that's how I understood the bullet point. If that was unclear, maybe:

```suggestion
  constrain Rust's unwinding implementation on any target.
```

is clearer ?

Reply:
@Centril I've adopted @gnzlbg's suggested wording.

Reply:
I'm happy with that.